Sources/Testing/Running/Configuration.TestFilter.swift

```diff
@@ -28,42 +28,52 @@ extension Configuration {
       case excluding
     }
 
-    /// An enumeration describing the various kinds of test filter.
-    fileprivate enum Kind: Sendable {
+    /// An enumeration describing a predicate that a test filter may specify.
+    fileprivate enum Predicate: Sendable, Codable {
       /// The test filter has no effect.
       ///
-      /// All tests are allowed when passed to a test filter with this kind.
-      case unfiltered
+      /// All tests are allowed when passed to a test filter with this
+      /// predicate.
+      case all
 
       /// The test filter contains a precomputed selection of test IDs.
       ///
       /// - Parameters:
       ///   - testIDs: The set of test IDs to predicate tests against.
       ///   - membership: How to interpret the result when predicating tests.
-      case precomputed(_ testIDs: Test.ID.Selection, membership: Membership)
+      case testIDs(_ testIDs: Set<Test.ID>, membership: Membership)
```

Comment:
By losing the precomputed selection and dropping down to a `Set`, we may be sacrificing a lot of performance. Can you confirm we're not recomputing the graph too frequently as a result of this change?

Reply:
We still use `Test.ID.Selection` later at the point where we apply the filter, so I believe the perf characteristics of this are unchanged